### PR TITLE
[app] Add Content Layout

### PR DESCRIPTION
**Pull request by zerefwayne**
**Description**

This PR defines the scaffolding layout for the app. It adds a responside side navigation with dummy links.

**Screenshots**

https://user-images.githubusercontent.com/28949397/124875334-667ef300-dfe6-11eb-9213-9872fc31302a.mp4

### CODE REVIEW

**Comment by zerefwayne** [comment]
For some reason, Codecov is not commenting on this pull request. Here's the [coverage report](https://codecov.io/gh/MahmoudAGawad/cros-diag-app/tree/aa3c6785738fec1135b94fdcfa48e2ba30b96695).

File: diagnostics-app/src/app/layout/header/header.component.it.spec.ts [hunk 1]
```diff
@@ -32,27 +31,21 @@ describe('integration: component: header', () => {
...
-    button = fixture.debugElement.query(By.css('button')).nativeElement;
-  });
-
-  it('service should emit event on click', () => {
-    spyOn(service, 'toggleTheme').and.callThrough();
-    expect(component.isDarkModeActivated()).toBeTrue();
-    button.click();
-    expect(service.toggleTheme).toHaveBeenCalled();
-    expect(component.isDarkModeActivated()).toBeFalse();
   });
```
Comment (MahmoudAGawad) [inline diagnostics-app/src/app/layout/header/header.component.it.spec.ts:34]: why removing this test?

File: diagnostics-app/src/app/layout/header/header.component.ts [hunk 2]
```diff
@@ -17,12 +17,18 @@ import { ThemeService } from 'src/app/core/services/theme.service';
...
+
   constructor(private _themeService: ThemeService) {}
 
   onToggleTheme() {
     this._themeService.toggleTheme();
   }
 
+  onToggleDrawer() {
+    this.toggleDrawer.emit();
+  }
```
Comment (MahmoudAGawad) [inline diagnostics-app/src/app/layout/header/header.component.ts:30]: we need to add tests for this method.

I am aware you are adding an integ. test in content-layout-component. But still, we need to test the behavior here.

Comment (MahmoudAGawad) [reply diagnostics-app/src/app/layout/header/header.component.ts:30]: oh, you already added a unit-test, that's fine then.

**Review by MahmoudAGawad — APPROVED** [review]
Overall looks good. LGTM % re-adding the removed test.

Thanks!

File: diagnostics-app/src/app/layout/header/header.component.it.spec.ts (hunk 1 above)
Comment (zerefwayne) [reply diagnostics-app/src/app/layout/header/header.component.it.spec.ts:34]: My bad, must've been removed during rebasing. Fixed it 👍🏻

**Pull request merged**